docs/cli/contrib.md

```diff
@@ -200,20 +204,35 @@ repository in one of the following ways:
   the `Manifest.json` file outside of the root directory, you must provide a
   `qooxdoo.json` file in the root dir (see below)
 
-### qooxdoo.json
+### Multi-library repositories / qooxdoo.json
 
-This file in the root of the repository allows the inclusion and discovery of 
-multiple libraries in this repo. It has the  following syntax:
+It is possible to put more than one library into a repository, by putting an
+index file into the root of the repository with the name "qooxdoo.json". 
+It allows the contrib system to auto-discover the contained libraries. 
+It has the  following syntax:
      
-````json5
+```json5
 {
   "libraries": [
    { "path":"relative-path/to/dir-containing-manifest1" },
-   { "path":"relative-path/to/dir-containing-manifest2" },
+   { "path":"relative-path/to/dir-containing-manifest2", main: true },
    //...
  ]
 }
-````
+```
+
+The first library will be treated as the main library, unless you specify the 
+main library by adding a truthy `main` property (see above.
+
+You can (re)generate the `qooxdoo.json` file by using 
+`qx contrib publish --create-index`. The command will automatically search for all
+`Manifest.json` files in the repository and ask you to select the 
+main library if you have more than one.
+
+Note that all libraries in the repository must have the same version number,
+because dependencies are managed and checked on the level of the repository, not
+of the library. When you `qx contrib publish` the repository, the versions of
+all libraries will be set to the one of the main library.
```

Comment:
hmmm this seems an odd restriction ... can you remind me why the contrib system must be able to handle this case. Why is it not enough to allow to have multiple libraries in a single git repo, but appart from that treate them as separate contribs ?

Reply:
The problem is - as we discussed earlier - that it would make the contrib system much more complex if you allowed multiple libraries to have divergent version numbers. You would have to add a separate lookup to the cache to store which version of which library is in which release of which repository. The rule that all libraries need to have the same version as the release name solves a lot of problems that we would have to address otherwise. We can think about a different system later, but for the moment it seems to make sense to me to keep it simple.